src/cdomains/apron/affineEqualityDomain.apron.ml

```diff
@@ -291,15 +308,21 @@ struct
 
   let meet t1 t2 =
     let sup_env = Environment.lce t1.env t2.env in
-    let t1, t2 = change_d t1 sup_env true false, change_d t2 sup_env true false
-    in if is_bot t1 || is_bot t2 then bot() else
+    let t1, t2 = change_d t1 sup_env true false, change_d t2 sup_env true false in
+    if is_bot t1 || is_bot t2 then 
+      bot ()
+    else
+      (* TODO: Why can I be sure that m1 && m2 are all Some here? *)
       let m1, m2 = Option.get t1.d, Option.get t2.d in
```

Comment:
Maybe it's supposed to be checking `is_bot_env` instead? I think that would guarantee it.

Reply:
That would guarantee it, though I am not sure if it is correct. I'll leave this TODO there for now.

Reply:
Then it'll fail if we ever encounter this case.